tests/unittests/tests-libc/tests-libc.c

```diff
@@ -75,6 +75,40 @@ static void test_libc_memchk(void)
     TEST_ASSERT(memchk(buffer, 0xff, sizeof(buffer)) == &buffer[5]);
 }
 
+static void test_libc_reverse_buf3(void)
+{
+    const char expected[3] = { 3, 2, 1 };
+    char buffer[3] = { 1, 2, 3 };
+    reverse_buf(buffer, sizeof(buffer));
+    TEST_ASSERT(!memcmp(buffer, expected, sizeof(buffer)));
+}
+
+static void test_libc_reverse_buf4(void)
+{
+    const char expected[4] = { 4, 3, 2, 1 };
+    char buffer[4] = { 1, 2, 3 , 4 };
+    reverse_buf(buffer, sizeof(buffer));
+    TEST_ASSERT(!memcmp(buffer, expected, sizeof(buffer)));
+}
+
+static void test_libc_dec_as_hex(void)
+{
+    uint32_t buf;
+    TEST_ASSERT_EQUAL_INT(1, dec_as_hex(12, &buf, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0x12, buf);
+
+    TEST_ASSERT_EQUAL_INT(2, dec_as_hex(123, &buf, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0x123, buf);
+
+    TEST_ASSERT_EQUAL_INT(3, dec_as_hex(123456, &buf, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0x123456, buf);
+
+    TEST_ASSERT_EQUAL_INT(4, dec_as_hex(12345678, &buf, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0x12345678, buf);
+
+    TEST_ASSERT_EQUAL_INT(-ENOBUFS, dec_as_hex(123456789, &buf, sizeof(buf)));
+}
```

Comment:
Misses testcases:
* Buffer of size zero
* How does the function behave with a pre-written buffer (okay, that case is implicit covered but please make it explicit)
* Zero input
* State of buffer when -ENOBUFS is returned

Reply:
Added, with the exception of 

> State of buffer when -ENOBUFS is returned


which I declared as undefined - if your buffer is not large enough to hold the result, expect your destination buffer to contain garbage.